src/main/java/frc/robot/Subsystem/Arm/Arm.java

```diff
@@ -0,0 +1,144 @@
+import com.ma5951.utils.StateControl.StatesTypes.StatesConstants;
+import com.ma5951.utils.StateControl.Subsystems.StateControlledSubsystem;
+import com.ma5951.utils.Utils.ConvUtil;
+
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import frc.robot.Subsystem.Arm.IOs.ArmIO;
+
+public class Arm extends StateControlledSubsystem {
+  private static Arm arm;
+
+  private ArmIO armIO = ArmConstants.getArmIO();
+  private double setPoint = 0;
+
+
+  private Pose3d armPosition;
+  private LoggedBool atPointLog;
+  private LoggedDouble setPointLog;
+  private LoggedDouble armAngleLog;
+  private LoggedDouble armOffset;
+  private LoggedPose3d armPose3d;
+  private LoggedBool CanMove;
+  private LoggedDouble feedForawdLog;
+
+
+  private pidControllerGainSupplier pidSupplier;
+
+  private Arm() {
+    super(ArmConstants.SUBSYSTEM_STATES , "Arm");
+    armPosition = ArmConstants.SIM_ARM_OFFSET;
+    atPointLog = new LoggedBool("/Subsystems/Arm/At Point");
+    setPointLog = new LoggedDouble("/Subsystems/Arm/Set Point");
+    armAngleLog = new LoggedDouble("/Subsystems/Arm/Arm Angle");
+    armOffset = new LoggedDouble("/Subsystems/Arm/Angle Offset");
+    armPose3d = new LoggedPose3d("/Subsystems/Arm/Position");
+    CanMove = new LoggedBool("/Subsystems/Arm/Can Move");
+    feedForawdLog = new LoggedDouble("/Subsystems/Arm/FeedForward");
+    armIO.setNutralMode(true);
+    pidSupplier = board.getPidControllerGainSupplier("Arm PID");
+    board.addNum("Angle Offset", 0);
+
+    board.addCommand("Reset Pose", new InstantCommand(() -> resetPosition(ArmConstants.ZERO_POSE)));
+    resetPosition(ArmConstants.ZERO_POSE);
+  }
+
+  public boolean isArmMoving() {
+    return armIO.getVelocity() > ArmConstants.kARM_MOVING_THRSHOLD_RPM;
+  }
+
+  public double getFeedForwardVoltage() {
+    return ((ArmConstants.MG * Math.sin(ConvUtil.DegreesToRadians(getArmPosition())) * ArmConstants.ARM_LENGTH) / ArmConstants.GEAR) / ArmConstants.kSTALL_TOURQE * 12;
+}
+
+  public double getCurrentDraw() {
+    return armIO.getCurrentDraw();
+  }
+
+  public void resetPosition(double position) {
+    armIO.resetPosition(position);
+  }
+
+  public boolean atPoint() {
+    return Math.abs(getArmPosition() - setPoint) <= ArmConstants.kTOLORANCE;
+  }
+
+  public double getArmPosition() {
+    return armIO.getPosition();
+  }
+
+  public double getSetPoint() {
+    return setPoint;
+  }
+
+  public void runSetPoint(double setPoint) {
+    this.setPoint = setPoint;
+    armIO.setAngleSetPoint(ConvUtil.DegreesToRotations(setPoint + board.getNum("Angle Offset")) , getFeedForwardVoltage());
+  }
+
+  public double getVoltage() {
+    return armIO.getAppliedVolts();
+  }
+
+  public void setVoltage(double voltage) {
+    armIO.setVoltage(voltage);
+  }
+
+  public void setPower(double power) {
+    armIO.setVoltage(power * 12);
+  }
+
+
+  //Can Move
+  private boolean LimitsCanMove(){
```

Comment:
```suggestion
  private boolean canMove(){
```
+ methods names should start with lower case